application/classes/Logger.php

```diff
@@ -9,41 +9,46 @@ public static function instance() {
 	}
 	
 	private static function loadLogger() {
-		$prov_config = Kohana::$config->load('logger');
+		if (is_null(Kohana::$config)) {
+			$prov_config = [ 'type' => 'syslog', 'config' => 'controll' ];
+		} else {
+			$prov_config = Kohana::$config->load('logger');
+		}
 		$fullclass = "Log_" . str_replace('/', '_', ucfirst($prov_config['type']));
-		$logger = new $fullclass($prov_config);
-		Kohana::$log->attach($logger);
+		$logger = new $fullclass(array_key_exists('config', $prov_config) ? $prov_config['config'] : $prov_config);
+		if (!is_null(Kohana::$log))
```

Comment:
This line implies a developer requires an intimate knowledge of state in order to properly initialize the logger - particularly if the singleton instance is present or not. Because this is an invariant, you may want to look into ways of enforcing the proper initialization of Kohana::$log instead of making a defensive check.